src/js/search/widgets/infinite_scroll.js

```diff
@@ -19,18 +19,24 @@ function renderTemplate(template, res) {
   return results;
 }
 
-function scrolledNearBottom(el, offset) {
-  return (el.scrollHeight - el.scrollTop) < offset;
+function scrolledNearBottom(scrollTop, offset) {
+  var bodyHeight = document.querySelector('body').clientHeight;
+  var pageHeight = document.documentElement.clientHeight;
+  return (bodyHeight - scrollTop - pageHeight) < offset;
```

Comment:
It makes sure the view port goes past the offset. It's a little clearer with a little sketch:

![elements](https://cloud.githubusercontent.com/assets/428280/23297583/5c67177c-fa7a-11e6-94ed-70764c6b438d.png)

> It looks like this will restrict it to only a full-page use.

It's true, it is restricted to a full page use. To make it "container-agnostic" we'd need:
* `bodyHeight` -> `contentHeight`
* `pageHeight` -> `containerHeight`

I can adapt that part, if you think it makes sense. If you have other ways of doing it, I'd be glad to hear!